contracts/gateway/L1GraphTokenGateway.sol

```diff
@@ -203,8 +203,8 @@ contract L1GraphTokenGateway is Initializable, GraphTokenGateway, L1ArbitrumMess
         bytes calldata _data
     ) external payable override notPaused returns (bytes memory) {
         IGraphToken token = graphToken();
-        require(_l1Token == address(token), "TOKEN_NOT_GRT");
         require(_amount != 0, "INVALID_ZERO_AMOUNT");
+        require(_l1Token == address(token), "TOKEN_NOT_GRT");
```

Comment:
Isn't it better to delay this check after the next line as well?

Reply:
Good catch, noted to be added in a future PR